plugins/foreman_openscap/0.12/index.md

```diff
@@ -0,0 +1,628 @@
+
+foreman_openscap does not create its own way to authenticate clients, but relies on already established certificate infrastructure when foreman_scap_client uploads reports to proxy. The proxy is either configured to use consumer certificates when your Foreman instance includes Katello, or Puppet certs when running vanilla Foreman. With Katello, consumer certificates are created for subscribed hosts and foreman_scap_client is able to use them to authenticate. Without Katello, Puppet certs are by default accepted by proxy when authenticating clients, which poses a problem when Ansible is used on hosts to deploy foreman_scap_client instead of Puppet - there are no certificates which client can use.
+
+  It is still possible to use Puppet CA generated certs to authenticate these clients. You need to generate the certificates yourself on your Puppet CA server:
+
+    puppet cert --generate ${client-fqdn}
+
+Copy the following certificates to your client:
+* /etc/puppetlabs/puppet/ssl/certs/ca.pem
+* /etc/puppetlabs/puppet/ssl/certs/${client-fqdn}.pem
+* /etc/puppetlabs/puppet/ssl/private_keys/${client-fqdn}.pem
+
+Configure appropriate Ansible variables of theforeman.foreman_scap_client with certificate paths on the client, for example:
+
+```
+foreman_scap_client_ca_cert_path: /etc/foreman_scap_client/ssl/certs/ca.pem
+foreman_scap_client_host_cert_path: /etc/foreman_scap_client/ssl/certs/${client-fqdn}.pem
+foreman_scap_client_host_private_key_path: /etc/foreman_scap_client/ssl/private_keys/${client-fqdn}.pem
+```
+
+Apply theforeman.foreman_scap_client role on your host, which will configure the cert paths for client and it will be able to authenticate.
+
+#### 4.6.2.2 Using manual policy deployment option
+
+If user decided to use manual option and your Foreman installation does not include Katello, then steps for generating and copying certificates from the previous section applies. It is necessary to add paths to certificates to /etc/foreman_scap_client/config.yaml.
+
+If your installation includes Katello, you need the consumer certificates:
+
+* /etc/rhsm/ca/katello-server-ca.pem
+* /etc/pki/consumer/cert.pem
+* /etc/pki/consumer/key.pem
+
+
+### 4.6.2 Manual policy deployments
+
+If a user chooses `Manual` as a policy deployment, Foreman will not care about client configuration and will leave it entirely up to user to configure clients. Multiple manual steps will be required to configure clients to send reports (which can be automated by using Ansible or Puppet).
+
+1) Install foreman_scap_client package on the client.
+
+2) Create client config in /etc/foreman_scap_client/config.yaml and populate it with appropriate values. See example config in section [6.3](plugins/foreman_openscap/0.9/index.html#6.3Configforclient). The most convenient way to get the attributes for policies is to add a host parameter with `<%= @host.policies_enc %>` value to your client and then get the JSON value it generates from ENC.
```

Comment:
we should provide API endpoint and hammer command to generate this more easily and/or possibly display this on host detail page, should I open RFE?

Reply:
Yes, it would be nice to have.

Reply:
opened https://projects.theforeman.org/issues/26324